tests/GrapheneOS/ConnectivityCheckSettingTest/src/grapheneos/test/ConnectivityCheckSettingTest.java

```diff
@@ -0,0 +1,143 @@
+        if (!getDevice().isAdbRoot()) {
+            try {
+                getDevice().enableAdbRoot();
+            } finally {
+                fail("adb root access is required");
+            }
+        }
+        getDevice().clearLogcat();
+    }
+
+    private final ArrayList<String> parseLogcat(String regexp, int setting) throws Exception {
+        final Pattern pattern = Pattern.compile(regexp);
+        ArrayList<String> result = new ArrayList<String>();
+
+        try (InputStreamSource logSource = getDevice().getLogcat()) {
+            InputStreamReader streamReader = new InputStreamReader(logSource.createInputStream());
+            BufferedReader logReader = new BufferedReader(streamReader);
+
+            String line;
+            while ((line = logReader.readLine()) != null) {
+                Matcher matcher = pattern.matcher(line);
+                if (setting != DISABLED_SETTING) {
+                    if (matcher.find() && line.contains(LOG_TAG)) {
+                        result.add(line);
+                        if (result.toString().contains("PROBE_HTTP") && result.toString().contains("PROBE_HTTPS")) {
+                            break;
+                        }
+                    }
+                } else {
+                    if (matcher.find()) {
+                        result.add(matcher.group(0));
+                        break;
+                    }
+                }
+            }
+            return result;
+        }
+    }
+
+    private final boolean checkParsed(ArrayList<String> matched, int setting) {
+        if (matched.size() <= 2) {
+            boolean result = switch (setting) {
+                case GRAPHENEOS_SERVER_SETTING ->
+                        matched.get(0).contains(ConnectivityCheckServer.GRAPHENEOS.httpUrl) && matched.get(1).contains(ConnectivityCheckServer.GRAPHENEOS.httpsUrl);
+                case STANDARD_SERVER_SETTING ->
+                        matched.get(0).contains(ConnectivityCheckServer.STANDARD.httpUrl) && matched.get(1).contains(ConnectivityCheckServer.STANDARD.httpsUrl);
+                case DISABLED_SETTING -> matched.get(0).equals(FILTER_DISABLED_REGEXP);
+                default -> false;
+            };
+            return result;
+        } else { return false; }
+    }
+
+    private boolean baseTest(int setting) throws Exception {
+        toggleWifi("disabled");
+        getDevice().setProperty(SYS_PROP, Integer.toString(setting));
+        RunUtil.getDefault().sleep(1000); // 1s
+        getDevice().clearLogcat();
+        toggleWifi("enabled");
+        RunUtil.getDefault().sleep(10000); // 10s
+        
+        return getDevice().checkConnectivity();
+    }
+
+    @Test
+    public void testGrapheneOSServer() throws Exception {
+        assertTrue(baseTest(GRAPHENEOS_SERVER_SETTING));
+        assertTrue(checkParsed(parseLogcat(FILTER_REQ_REGEXP, GRAPHENEOS_SERVER_SETTING), GRAPHENEOS_SERVER_SETTING));
```

Comment:
I think `checkParsed()` should be inlined here and in other tests to simplify the logic.